lib/Zonemaster/Backend/TestAgent.pm

```diff
@@ -60,12 +72,12 @@ sub run {
     $domain = $self->to_idn( $domain );
 
     if (defined $params->{ipv4} || defined $params->{ipv6}) {
-        Zonemaster::Engine->config->get->{net}{ipv4} = ( $params->{ipv4} ) ? ( 1 ) : ( 0 );
-        Zonemaster::Engine->config->get->{net}{ipv6} = ( $params->{ipv6} ) ? ( 1 ) : ( 0 );
+        Zonemaster::Engine::Profile->effective->set( q{net.ipv4}, ( $params->{ipv4} ) ? ( 1 ) : ( 0 ) );
+        Zonemaster::Engine::Profile->effective->set( q{net.ipv6}, ( $params->{ipv6} ) ? ( 1 ) : ( 0 ) );
     }
     else {
-        Zonemaster::Engine->config->get->{net}{ipv4} = 1;
-        Zonemaster::Engine->config->get->{net}{ipv6} = 1;
+        Zonemaster::Engine::Profile->effective->set( q{net.ipv4}, 1 );
+        Zonemaster::Engine::Profile->effective->set( q{net.ipv6}, 1 );
```

Comment:
Backend should either respect the `net.ipv4` and `net.ipv6` profile properties, or they shouldn't be part of the profile at all. Allowing the user to specify those values in their profile files just so Backend can ignore them isn't helpful in any way.